tests/__main__.py

```diff
@@ -136,22 +145,31 @@ def run_tests(argv, env, coverage=False):
             print('...coverage failed!')
             sys.exit(rc)
         print('...done')
+    elif junit_report_file:
+        os.environ.update(env)
+        with open(junit_report_file, 'wb') as output:
+            unittest.main(
+                testRunner=XMLTestRunner(output=output),
+                module=None,
+                argv=argv
+            )
     else:
         os.environ.update(env)
-        unittest.main(module=None, argv=argv)
-
+        unittest.main(module=None,argv=argv)
 
 if __name__ == '__main__':
```

Comment:
With the suggested ``runargs`` change, this would look like the following:

    runargs, lint= convert_argv(sys.argv[1:])
    fix_sys_path()
    if lint:
        check_lint()
    if runargs:
        argv, _, _, _ = runargs
        if '--start-directory' in argv:
            start = argv[argv.index('--start-directory') + 1]
            print('(will look for tests under {})'.format(start))

        run_tests(*runargs)